### PR TITLE
replace modal for removing dashcards with undo toast

### LINKED ISSUE #28736: Don't ask for confirmation when removing a dashboard card

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Closes https://github.com/metabase/metabase/issues/28736

### Description

To reduce friction, we are removing the confirmation modal that appears when a user attempts to remove a card from their dashboard. Now, clicking the "X" icon will immediately remove the card, but a toast will appear allowing the user to undo the removal.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Go to a dashboard in which you have edit access
2. Click pencil icon to edit dashboard -> mouseover any card -> click "X" icon to remove
3. Click "undo" to undo
4. Save, refresh page, and confirm card is still there

### Demo

https://user-images.githubusercontent.com/37751258/223232587-1b968a6b-e311-4367-8d25-efa2f4b4913d.mov

### Checklist

- ✅ Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/dashboard/components/DashboardGrid.jsx [hunk 1]
```diff
@@ -4,6 +4,8 @@ import PropTypes from "prop-types";
 
 import _ from "underscore";
 import cx from "classnames";
+import { connect } from "react-redux";
```
Comment (EmmadUsmani) [inline frontend/src/metabase/dashboard/components/DashboardGrid.jsx:7]: Start review here

File: frontend/src/metabase/dashboard/components/DashboardGrid.jsx [hunk 2]
```diff
@@ -253,7 +239,16 @@ class DashboardGrid extends Component {
...
+    this.props.removeCardFromDashboard({
+      dashcardId: dc.id,
+    });
+    this.props.addUndo({
+      message: t`Removed card`,
+      undo: true,
+      action: () =>
+        this.props.undoRemoveCardFromDashboard({ dashcardId: dc.id }),
+    });
+    MetabaseAnalytics.trackStructEvent("Dashboard", "Remove Card");
```
Comment (EmmadUsmani) [inline frontend/src/metabase/dashboard/components/DashboardGrid.jsx:251]: This method implementation is essentially the same as the one in the deleted `RemoveFromDashboardModal` except there is the additional `addUndo` call

File: frontend/src/metabase/dashboard/components/DashboardGrid.jsx [hunk 3]
```diff
@@ -253,7 +239,16 @@ class DashboardGrid extends Component {
   };
 
   onDashCardRemove(dc) {
-    this.setState({ removeModalDashCard: dc });
+    this.props.removeCardFromDashboard({
+      dashcardId: dc.id,
+    });
+    this.props.addUndo({
+      message: t`Removed card`,
```
Comment (alxnddr) [inline frontend/src/metabase/dashboard/components/DashboardGrid.jsx:246]: Adding a removed card name may be helpful. @mazameli what do you think?

Comment (EmmadUsmani) [reply frontend/src/metabase/dashboard/components/DashboardGrid.jsx:246]: Yeah I can add that, it should be a simple change

**Review by ranquild — APPROVED** [review]

**Pull request merged**